### ISSUE TITLE
Craft Control Panel doesn't log user out after session timeout

### ISSUE THREAD

**Issue by Anubarak** [craftcms/cms#3951]
### Description
The `AuthManager.js` doesn't log the user out after it displays that the `LogoutWarningModal` via `this.showLogoutWarningModal()`, it actually only shows a modal and that's basically it. 
You can even delete the modal via Chrome Dev tools and still create entries. When you set the `userSessionDuration` to something low (like 40 seconds) it keeps cycling between the "your session is expired" and "your session will timeout" modals without doing anything.

### Steps to reproduce

1. set `userSessionDuration` to something low, so you can wait until the logic runs (eg 40s)
2. log into the CP and wait
3. it will show both modals again and again
4. go to your dev tools and remove both modals, you can nagivage in your CP as you like
5. do whatever you want, your session is not expired, you can even store entries but still displays both modals whenever you change the URL/site

You can see a video here [click](https://drive.google.com/file/d/1HUROgF6VGowyEAEgdTJ2z6JqBiqScYA6/view?usp=sharing) (it's a link to my google drive)

I "fixed" it by making an ajax call to `/logout` after showing the window

**Comment by brandonkelly** [comment]
Do you have any user session-related config settings set, besides `userSessionDuration` ?

**Comment by Anubarak** [comment]
No I have not. I changed nothing else besides this setting for testing purposes. And it doesn't matter if I set the value so something small as 40 or something larger that's higher than the 120s interval in the AuthManager. As long as the session isn't expired by default the behavior stays the same even when I set it to 500/1000 and so on. I'll check if it even happens with the default 3600s tomorrow but each test consumes much time

**Comment by brandonkelly** [comment]
So to be clear, that modal is not responsible for actually logging you out. It’s just there to let you know when your session is about to expire, and if you have logged out, give you a chance to log back in so you don’t lose any in-progress content.

If it’s showing the login form, then as far as it can tell, you actually have been logged out at that point. (That is based on a `users/get-remaining-session-time` request that comes back with `0`.)

So the mystery here is why is Craft responding to the JS saying that you’re logged out for that `users/get-remaining-session-time` request, but then on a subsequent request it thinks you’re still logged in.

**Comment by Anubarak** [comment]
@brandonkelly Maybe it's because of this setting [$authTimeout](https://www.yiiframework.com/doc/api/2.0/yii-web-user#$authTimeout-detail)

> The number of seconds in which the user will be logged out automatically if he remains inactive. If this property is not set, the user will be logged out after the current session expires (c.f. yii\web\Session::$timeout). **Note that this will not work if $enableAutoLogin is true.**

`authTimeout` is set to true by default in Craft
```PHP
    public static function userConfig(): array
    {
        $configService = Craft::$app->getConfig();
        $generalConfig = $configService->getGeneral();
        $request = Craft::$app->getRequest();

        if ($request->getIsConsoleRequest() || $request->getIsSiteRequest()) {
            $loginUrl = UrlHelper::siteUrl($generalConfig->getLoginPath());
        } else {
            $loginUrl = UrlHelper::cpUrl('login');
        }

        $stateKeyPrefix = md5('Craft.' . WebUser::class . '.' . Craft::$app->id);

        return [
            'class' => WebUser::class,
            'identityClass' => User::class,
            'enableAutoLogin' => true,  // <--
            'autoRenewCookie' => true,
            'loginUrl' => $loginUrl,
            'authTimeout' => $generalConfig->userSessionDuration ?: null,
            'identityCookie' => Craft::cookieConfig(['name' => $stateKeyPrefix . '_identity']),
            'usernameCookie' => Craft::cookieConfig(['name' => $stateKeyPrefix . '_username']),
            'idParam' => $stateKeyPrefix . '__id',
            'tokenParam' => $stateKeyPrefix . '__token',
            'authTimeoutParam' => $stateKeyPrefix . '__expire',
            'absoluteAuthTimeoutParam' => $stateKeyPrefix . '__absoluteExpire',
            'returnUrlParam' => $stateKeyPrefix . '__returnUrl',
        ];
    }
```

As soon as I overwrite this property to `false` it works as expected otherwise the modals are just useless (even thought they are really pretty) because it doesn't expire the session. They'll only work if the `userSessionDuration` is not changed

**Comment by jeffers102** [comment]
In my case, I receive the modal dialog saying my session will expire in 2 minutes.  After it counts down to 0, I am immediately logged back in and automatically.  In theory, I could walk away from my machine and my session would never expire.  When I set "autoRenewCookie" to false in App.php, it no longer performs this action.  Are there any plans to patch Craft by at least allowing overrides in the config file for these two settings (enableAutoLogin and autoRenewCookie)?

**Comment by brandonkelly** [comment]
@jeffers102 We are able to reproduce that if the `identity` cookie was originally created while the `userSessionDuration` was set to a higher number than it is currently set to. Looks like Yii is using the same session duration value for the cookie duration as well, and once it’s set on the cookie, it will continue using the same cookie-based value going forward (unless the cookie expires). Maybe that explains what you are seeing.

It is possible to disable identity cookies from `config/app.web.php`:

```php
<?php

return [
    'components' => [
        'user' => function() {
            $config = craft\helpers\App::userConfig();
            $config['autoRenewCookie'] = false;
            return Craft::createObject($config);
        },
    ],
];
```

**Comment by brandonkelly** [comment]
I finally got to the bottom of this issue. Each time Craft was checking to see how much time is left on the user session, those requests were renewing the identity cookie. So if you stay on a single CP page long enough for your session to time out, your session cookie would still be active, so simply reloading the page could end up logging you back in via that cookie.

Under normal circumstances, the cookie and the PHP session are supposed to be in sync, so they both expire at the same time, and the cookie is only meant as a backup in case you lose the PHP session (e.g. you quit your browser).

Fix will be in the next 3.3 release.
